### DISCUSSION TITLE
BaseInFilter alternate delimiter?

### DISCUSSION THREAD

**Discussion by kj-asrc** [carltongibson/django-filter#1414] [Q&A] [answered]
Hi,

First off, thank you to those working on this awesome module, it's been super helpful!

My issue is mostly just a question/idea for the BaseInFilter class. Is it currently possible to change the delimiter of the BaseInFilter? I see that the class itself is a child class of the BaseCSVFilter, so I'm guessing it's not. Is there any support or plans to support using a different delimiter and/or escaping the comma?

Thank you for your time!

**Comment by Plonq** [comment]
I'm wondering the same thing. Using `BaseInFilter` with `CharFilter` means that your values can't have `,` in them. I'd like a way to escape the comma, or do the equivalent of quoting the values.

The following don't work (not that I really expected them to):

* URL Encoding the comma: `in_filter=ab%2Ccd`
* Adding quotes: `in_filter="ab,cd"`
* Backslash: `in_filter="ab\,cd"`

**Comment by carltongibson** [comment, marked as answer]
The CSV filters were added for API usage. The previous approach of multiple entries still works, and I’d say go for that. 

`?a=1&a=2&a=3`

Either that or you’ll need to adapt the CSV code to your own needs, but it’s not something I’d want to include.

**Comment by realsuayip** [comment]
@kj-asrc 

It is actually possible to achieve this using `django.contrib.postgres.forms.SimpleArrayField`

```python
class ArrayFilter(filters.Filter):
    field_class = SimpleArrayField
```

and then something like this:

```python
ids = ArrayFilter(
    field_name="id",
    lookup_expr="in",
    base_field=forms.IntegerField(min_value=0),
    max_length=50,
    delimiter=",", # <-- here
    required=True,
    help_text="Multiple values may be separated by commas. Enter 50 at most.",
)
```

You can even specify `max_length` and `min_length`, which are not available via `BaseCSVFilter` the error message also points to the sequence of item that failed to validate.
